app/src/main/java/com/dimagi/biometric/fragments/BaseMatchFragment.java

```diff
@@ -0,0 +1,162 @@
+package com.dimagi.biometric.fragments;
+
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.content.pm.PackageManager;
+import android.net.Uri;
+import android.os.Bundle;
+import android.Manifest;
+
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.activity.result.contract.ActivityResultContracts;
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AlertDialog;
+import androidx.core.content.ContextCompat;
+import androidx.fragment.app.Fragment;
+
+import android.provider.Settings;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.TextView;
+
+import com.dimagi.biometric.R;
+
+/**
+ * A base class to contain common functionality for both the face and finger match fragments.
+ */
+public abstract class BaseMatchFragment extends Fragment {
+    private static final String TAG = "BIOMETRIC";
+
+    private final String PREV_RATIONALE_KEY = "prev_rationale";
+    private final String DIALOG_STATE_KEY = "dialog_state";
+
+    private DialogState dialogState = DialogState.NONE;
+
+    private enum DialogState {
+        NONE,
+        RATIONALE,
+        SETTINGS
+    }
+
+    protected abstract void handleStartCapture();
+    protected abstract void handleCancelCapture();
+
+    public BaseMatchFragment() {
+        // Required empty public constructor
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_main_menu, container, false);
+        Button startButton = view.findViewById(R.id.start_capture_button);
+        startButton.setOnClickListener(v -> {
+            if (ContextCompat.checkSelfPermission(
+                    requireContext(), Manifest.permission.CAMERA) == PackageManager.PERMISSION_GRANTED) {
+                handleStartCapture();
+            } else {
+                if (shouldShowRequestPermissionRationale(Manifest.permission.CAMERA)) {
+                    savePreviousRationale();
+                }
+                if (!isPermissionDeniedPermanently()) {
```

Comment:
@zandre-eng It seems that nothing happens when the permission has been permanently denied, maybe we should consider providing the user some sort of justification.

Reply:
I have implemented requesting an additional required permission (commcare read) and as a part of these changes I have accounted for showing the user a dialog to navigate them to the app settings page if all permissions are permanently denied ([678e85f](https://github.com/dimagi/biometric-android/pull/1/commits/678e85fb4c5927f90991f4e81a7a441adeb0a1ee)).